### PR TITLE
Frame sim

### PR DESCRIPTION
**Pull request by wnichols1**

### CODE REVIEW

File: data/frameSimulator/src/PercivalFrameSimulatorPlugin.cpp [hunk 1]
```diff
@@ -0,0 +1,220 @@
+
+
+#include "version.h"
+#include "../include/PercivalFrameSimulatorPlugin.h"
```
Comment (ajgdls) [inline data/frameSimulator/src/PercivalFrameSimulatorPlugin.cpp]: Are you sure this needs to be a relative include path here?  What happens when these files get installed?

File: data/frameSimulator/src/checkframe.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import os
+import re
+import struct
+import re;
+import h5py;
+import numpy as np;
+
+
+
+f = h5py.File('/tmp/haha_000001.h5','r')
```
Comment (ajgdls) [inline data/frameSimulator/src/checkframe.py]: What happens when this file does not exist?  Should this be a command line option?

File: data/frameSimulator/src/checkframe.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        else:
+            if (fr[row,col]!=0 and fr[row,col] != counter):
+                print "pixel bad at ", row, col;
+
+        counter = counter + 1;
+        if(counter == 65536):
+            counter = 0;
+        
+    i = i+1;
+
```
Comment (ajgdls) [inline data/frameSimulator/src/checkframe.py]: Does this checkframe only work with your simulator (I assume yes)?  Perhaps you should make this clear either in the name of the file or with some comments at the beginning.

**Review by ajgdls — COMMENTED** [review]
Look OK, I haven't built it but I assume that bit is working.  Just wondering about the apparent relative include that you have.  Maybe you need to install that header file and then you can simply include the file without the relative path.

**Review by ajgdls — APPROVED** [review]

File: data/frameSimulator/src/PercivalFrameSimulatorPlugin.cpp (hunk 1 above)
Comment (wnichols1) [reply data/frameSimulator/src/PercivalFrameSimulatorPlugin.cpp]: fixed.

File: data/frameSimulator/src/checkframe.py (hunk 3 above)
Comment (wnichols1) [reply data/frameSimulator/src/checkframe.py]: yes I shall rename it and put a comment at the top. I don't fancy doing command-line options although I realize it's a possibility. Maybe in the next draft.

**Pull request merged**